draft-irtf-cfrg-rsa-blind-signatures.md

```diff
@@ -380,175 +371,90 @@ Steps:
-~~~
-
-### SaltedFinalize
-
-SaltedFinalize invokes Finalize directly with the salted
-message and outputs the result.
-
-~~~
-SaltedFinalize(pkS, msg, msg_salt, blind_sig, inv)
-
-Parameters:
-- kLenInBytes, the length in bytes of the RSA modulus n
-
-Inputs:
-- pkS, server public key (n, e)
-- msg, message to be signed, an byte string
-- msg_salt, the 32 bytes random salt used to salt the message
-- blind_sig, signed and blinded element, an byte string of
-  length kLenInBytes
-- inv, inverse of the blind, an integer
-
-Outputs:
-- sig, an byte string of length kLenInBytes
-
-Errors:
-- "invalid signature": Raised when the signature is invalid
-- "unexpected input size": Raised when a byte string input doesn't
-  have the expected length.
-
-Steps:
-1. salted_msg = msg_salt || msg
-2. output Finalize(pkS, salted_msg, blind_sig, inv)
-~~~
-
-### SaltedVerify
+# Message Randomization {#randomization}
 
-SaltedVerify validates the resulting unblinded signature computed over a
-salted message. It invokes RSASSA-PSS-VERIFY directly by augmenting the input
-message with the message salt.
+Message randomization is the process by which the message to be signed and verified
+is augmented with fresh randomness. We denote this process by the function Randomize(msg),
+which takes as input a message `msg` and produces a randomized message `randomMsg`.
+Its implementation is shown below.
 
 ~~~
-SaltedVerify(pkS, msg, msg_salt, sig)
-
-Parameters:
-- kLenInBytes, the length in bytes of the RSA modulus n
+Randomize(msg)
 
 Inputs:
-- pkS, server public key (n, e)
-- msg, message to be signed, an byte string
-- msg_salt, the 32 bytes random salt used to salt the message
-- sig, signature of the salted_msg
+- msg, message to be signed, a byte string
 
 Outputs:
-- "valid signature" if the signature is valid
-
-Errors:
-- "invalid signature": Raised when the signature is invalid
+- randomMsg, a byte string that is 32 bytes longer than input msg
 
 Steps:
-1. salted_msg = msg_salt || msg
-2. result = RSASSA-PSS-VERIFY(pkS, salted_msg, sig)
-3. If result = "valid signature", output "valid signature", else
-  raise "invalid signature" and stop
+1. msgSalt = random(32)
+2. randomMsg = msgSalt || msg
+3. output randomMsg
 ~~~
 
-## Encoding Options {#pss-options}
+# RSABSSA Variants {#rsabssa}
 
-The RSASSA-PSS parameters, defined as in {{!RFC8017, Section 9.1.1}}, are as follows:
+In this section we define different named variants of RSABSSA. Each variant specifies
+a hash function, RSASSA-PSS parameters as defined in {{!RFC8017, Section 9.1.1}}, and
+whether or not message randomization (as described in {{randomization}}) is performed
+on the input message. Future specifications can introduce other variants as desired.
+The named variants are as follows:
 
-- Hash: hash function
-- MGF: mask generation function
-- sLenInBytes: intended length in bytes of the salt
+1. RSABSSA-SHA384-PSS-Randomized: This named variant uses SHA-384 as the hash function,
```

Comment:
it is not exactly clear in the draft at which point Randomize(msg) is invoked. The diagram at the beginning could display it.
Is it invoked before Blind or as part of Blind?

Reply:
The diagram shows only the core protocol, which does _not_ include randomization. Message randomization is invoked before the core protocol. I'll clarify this in the text.

Reply:
Done!